### PR TITLE
doc(rjy): polish d4pg/ppg/qrdqn policy doc

### PR DESCRIPTION
**Pull request by nighood**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/policy/qrdqn.py [hunk 1]
```diff
@@ -15,7 +15,9 @@
 class QRDQNPolicy(DQNPolicy):
     r"""
     Overview:
-        Policy class of QRDQN algorithm.
+        Policy class of QRDQN algorithm. QRDQN is a distributional RL algorithm, which is an extension of DQN. \
+        The main idea of QRDQN is to use quantile regression to estimate the quantile of the distribution of \
+        the return value, and then use the quantile to calculate the quantile loss.
```
Comment (PaParaZz1) [inline ding/policy/qrdqn.py:20]: add paper link

File: ding/policy/d4pg.py [hunk 2]
```diff
@@ -149,13 +151,34 @@ class D4PGPolicy(DDPGPolicy):
     )
 
     def default_model(self) -> Tuple[str, List[str]]:
+        """
+        Overview:
+            Return this algorithm default neural network model setting for demonstration. ``__init__`` method will \
```
Comment (zjowowen) [inline ding/policy/d4pg.py:156]: "Return this algorithm default neural network model setting for demonstration." -> "Return the default neural network model class for D4PGPolicy."

File: ding/policy/d4pg.py [hunk 3]
```diff
@@ -204,13 +227,35 @@ def _init_learn(self) -> None:
...
+                often need to first be stacked in the batch dimension by some utility functions such as \
+                ``default_preprocess_learn``. \
+                For D4PG, each element in list is a dict containing at least the following keys: ``obs``, \
+                ``action``, ``reward``, ``next_obs``. Sometimes, it also contains other keys such as ``weight``.
+
         Returns:
-            - info_dict (:obj:`Dict[str, Any]`): Including at least actor and critic lr, different losses.
+            - info_dict (:obj:`Dict[str, Any]`): The output result dict of forward learn, \
+                containing at least the "actor lr", "critic lr", "different losses", "q_value", "action", "priority", \
+                keys. Additionally, loss_dict also contain other keys, which are mainly used for monitoring and \
```
Comment (zjowowen) [inline ding/policy/d4pg.py:248]: "Additionally, loss_dict also contain other keys, which are mainly used for monitoring and" -> "Additionally, loss_dict also contains other keys, which is mainly used for monitoring and"

File: ding/policy/ppg.py [hunk 4]
```diff
@@ -14,6 +14,17 @@
 
 
 class ExperienceDataset(Dataset):
+    """
+    Overview:
+        A dataset class for storing and accessing experience data.
+
+    Arguments:
```
Comment (zjowowen) [inline ding/policy/ppg.py:21]: Put Arguments into the notation of def __init__ function.

File: ding/policy/ppg.py [hunk 5]
```diff
@@ -14,6 +14,17 @@
...
+        data (:obj:`dict`): A dictionary containing the experience data, where the keys represent the data types \
```
Comment (zjowowen) [inline ding/policy/ppg.py:22]: "data (:obj:`dict`)" -> "- data (:obj:`dict`)"

File: ding/policy/ppg.py [hunk 6]
```diff
@@ -615,11 +709,15 @@ def learn_aux(self) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
 class PPGOffPolicy(Policy):
     """
     Overview:
-        Policy class of PPG algorithm.
+        Policy class of PPG algorithm with off-policy training mode. We implement off-policy PPG with two buffers \
```
Comment (zjowowen) [inline ding/policy/ppg.py:712]: "We implement off-policy PPG with two buffers with different data use constraint (max_use), which policy buffer offers data for policy phase while value buffer provides auxiliary phase's data." -> "Off-policy PPG contains two different data max_use buffers.  The policy buffer offers data for policy phase , while the value buffer provides auxiliary phase's data."

File: ding/policy/ppg.py [hunk 7]
```diff
@@ -792,30 +899,37 @@ def _forward_learn(self, data: dict) -> Dict[str, Any]:
         Overview:
             Forward and backward function of learn mode.
         Arguments:
-            - data (:obj:`Dict[str, Any]`): Dict type data, a batch of data for training, values are torch.Tensor or \
-                np.ndarray or dict/list combinations.
+            - data (:obj:`Dict[str, Any]`): Input data used for policy forward, including the \
+                collected training samples from replay buffer. For each element in dict, the key of the \
+                dict is the name of data items and the value is the corresponding data. Usually, the value is \
```
Comment (zjowowen) [inline ding/policy/ppg.py:904]: "Usually, the value is ..." ->
"Usually, the class type of value is either torch.Tensor or np.ndarray, or a dict/list containing either torch.Tensor or np.ndarray items"

File: ding/policy/d4pg.py [hunk 8]
```diff
@@ -204,13 +228,35 @@ def _init_learn(self) -> None:
...
+                torch.Tensor or np.ndarray or there dict/list combinations. In the ``_forward_learn`` method, data \
+                often need to first be stacked in the batch dimension by some utility functions such as \
+                ``default_preprocess_learn``. \
+                For D4PG, each element in list is a dict containing at least the following keys: ``obs``, \
+                ``action``, ``reward``, ``next_obs``. Sometimes, it also contains other keys such as ``weight``.
+
         Returns:
-            - info_dict (:obj:`Dict[str, Any]`): Including at least actor and critic lr, different losses.
+            - info_dict (:obj:`Dict[str, Any]`): The output result dict of forward learn, \
+                containing at least the "actor lr", "critic lr", "different losses", "q_value", "action", "priority", \
```
Comment (PaParaZz1) [inline ding/policy/d4pg.py:248]: cur_lr_actor

File: ding/policy/d4pg.py [hunk 9]
```diff
@@ -204,13 +228,35 @@ def _init_learn(self) -> None:
         self._forward_learn_cnt = 0  # count iterations
 
     def _forward_learn(self, data: dict) -> Dict[str, Any]:
```
Comment (PaParaZz1) [inline ding/policy/d4pg.py:230]: `data: List[Dict[str, Any]]`  refer to dqn/ddpg

File: ding/policy/qrdqn.py [hunk 10]
```diff
@@ -125,14 +146,38 @@ def _init_learn(self) -> None:
...
+        """
         Overview:
-            Forward and backward function of learn mode.
+            Policy forward function of learn mode (training policy and updating parameters). Forward means \
+            that the policy inputs some training batch data from the replay buffer and then returns the output \
+            result, including various training information such as loss, current lr.
+
         Arguments:
-            - data (:obj:`dict`): Dict type data, including at least ['obs', 'action', 'reward', 'next_obs']
+            - data (:obj:`dict`): input data used for policy forward, including the \
```
Comment (PaParaZz1) [inline ding/policy/qrdqn.py:156]: Input

File: ding/policy/ppg.py [hunk 11]
```diff
@@ -357,10 +394,16 @@ def _load_state_dict_learn(self, state_dict: Dict[str, Any]) -> None:
         self._optimizer_aux_critic.load_state_dict(state_dict['optimizer_aux_critic'])
 
     def _init_collect(self) -> None:
-        r"""
+        """
         Overview:
-            Collect mode init method. Called by ``self.__init__``.
-            Init unroll length, collect model.
+            Initialize the collect mode of policy, including related attributes and modules. For PPO, it contains the \
```
Comment (PaParaZz1) [inline ding/policy/ppg.py:399]: for PPG

File: ding/policy/ppg.py [hunk 12]
```diff
@@ -396,14 +457,21 @@ def _forward_collect(self, data: dict) -> dict:
     def _process_transition(self, obs: Any, model_output: dict, timestep: namedtuple) -> dict:
         """
         Overview:
-               Generate dict type transition data from inputs.
+               Process and pack one timestep transition data into a dict, which can be directly used for training and \
```
Comment (PaParaZz1) [inline ding/policy/ppg.py:460]: wrong indent

File: ding/policy/ppg.py [hunk 13]
```diff
@@ -396,14 +457,21 @@ def _forward_collect(self, data: dict) -> dict:
...
+            saved in replay buffer. For PPG, it contains obs, next_obs, action, reward, done, logit, value.
         Arguments:
                 - obs (:obj:`Any`): Env observation
-                - model_output (:obj:`dict`): Output of collect model, including at least ['action']
-                - timestep (:obj:`namedtuple`): Output after env step, including at least ['obs', 'reward', 'done']\
-                       (here 'obs' indicates obs after env step).
+                - model_output (:obj:`dict`): The output of the policy network with the observation \
+                as input. For PPG, it contains the state value, action and the logit of the action.
```
Comment (PaParaZz1) [inline ding/policy/ppg.py:465]: there should be some indents

File: ding/policy/ppg.py [hunk 14]
```diff
@@ -417,13 +485,20 @@ def _process_transition(self, obs: Any, model_output: dict, timestep: namedtuple
         return transition
 
     def _get_train_sample(self, data: list) -> Union[None, List[Any]]:
```
Comment (PaParaZz1) [inline ding/policy/ppg.py:487]: typing lint

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**